pkg/registry/core/pod/storage/eviction.go

```diff
@@ -104,40 +104,85 @@ func propagateDryRun(eviction *policy.Eviction, options *metav1.CreateOptions) (
 func (r *EvictionREST) Create(ctx context.Context, name string, obj runtime.Object, createValidation rest.ValidateObjectFunc, options *metav1.CreateOptions) (runtime.Object, error) {
 	eviction, ok := obj.(*policy.Eviction)
 	if !ok {
-		return nil, errors.NewBadRequest(fmt.Sprintf("not a Eviction object: %T", obj))
+		return nil, kerrors.NewBadRequest(fmt.Sprintf("not a Eviction object: %T", obj))
 	}
 
 	if name != eviction.Name {
-		return nil, errors.NewBadRequest("name in URL does not match name in Eviction object")
+		return nil, kerrors.NewBadRequest("name in URL does not match name in Eviction object")
 	}
 
-	deletionOptions, err := propagateDryRun(eviction, options)
+	originalDeleteOptions, err := propagateDryRun(eviction, options)
 	if err != nil {
 		return nil, err
 	}
 
-	obj, err = r.store.Get(ctx, eviction.Name, &metav1.GetOptions{})
-	if err != nil {
-		return nil, err
-	}
-	pod := obj.(*api.Pod)
-
 	if createValidation != nil {
 		if err := createValidation(ctx, eviction.DeepCopyObject()); err != nil {
 			return nil, err
 		}
 	}
 
-	// Evicting a terminal pod should result in direct deletion of pod as it already caused disruption by the time we are evicting.
-	// There is no need to check for pdb.
-	if pod.Status.Phase == api.PodSucceeded || pod.Status.Phase == api.PodFailed {
+	var pod *api.Pod
+	deletedPod := false
+	var unrecoverableConflict *error
+	err = retry.RetryOnConflict(EvictionsRetry, func() error {
```

Comment:
would this be simpler:

```go
// by default, retry conflict errors
shouldRetry := kerrors.IsConflict
if !resourceVersionIsUnset(originalDeleteOptions) {
	// if the original options included a resourceVersion precondition, don't retry
	shouldRetry = func(err error) bool {return false}
}

err = retry.OnError(EvictionsRetry, shouldRetry, func() error {
```

Reply:
That retry library is handy.  Great suggestion.